source/agora/test/Base.d

```diff
@@ -907,7 +907,102 @@ public class TestAPIManager
             ex.line = line;
             throw ex;
         }
+        retryFor(nodes[0].client.getBlocksFrom(0, 1).length > 0, 2.seconds);
     }
+
+    /*******************************************************************************
+
+        Add blocks up to the provided height
+
+        Params:
+            target_height = the target height after adding blocks.
+            enrolled_height = the height with the active enrollments
+
+    *******************************************************************************/
+
+    public void generateBlocks (Height target_height, Height enrolled_height = Height(0),
+        string file = __FILE__, size_t line = __LINE__)
+    {
+        generateBlocks(clients.take(GenesisValidators), target_height, enrolled_height,
+            file, line);
+    }
+
+    /// Ditto
+    public void generateBlocks (Clients)(Clients clients,
```

Comment:
How about merging the two overriden functions into one function with the `clients` parameter having a default `null` parameter. If `null` is passed in, it can `use this.clients`. Please think about it.

Reply:
I will try this out. There are other default parameters already so it depends if this will make it less convenient.

Reply:
@linked0 It was a good idea. 
I have combined them and used `this.clients` as default.